persistent/test/Database/Persist/THSpec.hs

```diff
@@ -76,6 +77,10 @@ import qualified Database.Persist.TH.TypeLitFieldDefsSpec as TypeLitFieldDefsSpe
 -- machinery
 type TextId = Text
 
+-- | Location above the block defining Person below. Must be before it. Do not move!
+-- Used to test TH definition positions are plausible.
+personDefBeforeLoc :: SourceLoc
+personDefBeforeLoc = $(TH.lift @_ @SourceLoc . sourceLocFromTHLoc =<< TH.location)
```

Comment:
I think there may be a better way of doing this - 

```haskell
do
  firstLoc <- [d| firstLocation = $(TH.location) |]
  decls <- share [mkPersistWith sqlSettings ...] [persistUpperCase| ... |]
  secondLoc <- [d| secondLocation = $(TH.location) |]
  pure (concat [firstLoc, decls, secondLoc])
```

This would require a little less spooky magic

Reply:
Ah good point. My apologies

It looks like this is failing on GHC 8.8 - complaining about `TH.lift @_ @SourceLoc`. My guess is that GHC 8.8 has only one type variable for `lift`